contracts/RentingV3.vy

```diff
@@ -479,36 +480,35 @@ def revoke_listing(token_contexts: DynArray[TokenContext, 32]):
 
 
 @external
-def start_rentals(token_contexts: DynArray[TokenContextAndListing, 32], duration: uint256, delegate: address, signature_timestamp: uint256):
+def start_rentals(token_contexts: DynArray[TokenContextAndListing, 32], delegate: address, signature_timestamp: uint256):
 
     """
     @notice Start rentals for multiple NFTs for the specified duration and delegate them to a wallet
     @dev Iterates over token contexts to begin rentals for each NFT. The rental conditions are evaluated against the matching listing, signed by the owner and the protocol admin. The rental amount is computed and transferred to the protocol wallet and the delegation is created for the given wallet.
     @param token_contexts An array of token contexts, each containing the rental state and signed listing for an NFT.
-    @param duration The duration of the rentals in hours.
     @param delegate The address to delegate the NFT to during the rental period.
     @param signature_timestamp The timestamp of the protocol admin signature.
     """
 
     self._check_not_paused()
 
     rental_logs: DynArray[RentalLog, 32] = []
-    expiration: uint256 = block.timestamp + duration * 3600
     rental_amounts: uint256 = 0
 
     for context in token_contexts:
-        rental_amounts += self._compute_rental_amount(block.timestamp, expiration, context.signed_listing.listing.price)
+        rental_amounts += self._compute_rental_amount(block.timestamp, block.timestamp + context.duration * 3600, context.signed_listing.listing.price)
```

Comment:
does it make sense to define the `expiration` before this so we don't compute the `block.timestamp + context.duration * 3600` twice?

Reply:
the expiration is now context dependent because of the duration. we could define it upfront as an array, but that would cost more than the computation. you see other way?